pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -658,9 +657,16 @@ public boolean equals(Object o) {
   }
 
   /**
-   * Check whether the current schema is backward compatible with oldSchema.
+   * Check whether the current schema is backward compatible with the given old schema. The current schema can be
+   * modified in-place to match the old schema.
```

Comment:
I don't particularly like that a method called `isCompatible` modifies the schema. I would rather have another method `updateBooleanIfNeeded`, and then call isCompatible() as a read-only API.

Reply:
Good point, updated the PR